### PR TITLE
refactor : enhance API /substrate/countries

### PR DESCRIPTION
**Pull request by rubenkristian**
### JIRA LINK
- https://blocksphere2020.atlassian.net/browse/DBIO-2338

### Changelog
- update getAggregatedByCountries
- update unit test service-request service

### CODE REVIEW

**Comment by rubenkristian** [comment]
image

**Review by hilyds — APPROVED** [review]

**Review by agustinustheo — APPROVED** [review]

File: src/endpoints/substrate-endpoint/services/service-request.service.ts [hunk 1]
```diff
@@ -7,6 +7,27 @@ interface RequestsByCountry {
...
+  city: string;
+  totalRequests: number;
+  totalValue: CurrencyValueInterface;
+}
+interface RequestByCountryDictInterface {
+  totalRequests: number;
+  totalValue: string | number;
+  services: {
+    [id: string]: ServiceInterface;
+  };
```
Comment (agustinustheo) [inline src/endpoints/substrate-endpoint/services/service-request.service.ts:30]: Please separate this into separate files. `service-request.interfaces.ts`

File: src/endpoints/substrate-endpoint/services/service-request.service.ts [hunk 2]
```diff
@@ -64,114 +73,91 @@ export class ServiceRequestService {
...
-            city: request.city,
-            totalRequests: 0,
-            totalValue: {
-              dbio: 0,
-              dai: 0,
-              usd: 0,
-            },
-          };
+          requestByCountryDict[country]['services'][
+            service.region + '-' + service.city + '-' + service.category
```
Comment (agustinustheo) [inline src/endpoints/substrate-endpoint/services/service-request.service.ts:120]: Please turn this into its own variable somewhere to reuse.

**Review by agustinustheo — CHANGES_REQUESTED** [review]

File: src/endpoints/substrate-endpoint/services/service-request.service.ts (hunk 2 above)
Comment (rubenkristian) [reply src/endpoints/substrate-endpoint/services/service-request.service.ts]: done

**Pull request merged**